### DISCUSSION TITLE
[Extended Sync] How to model parallel access?

### DISCUSSION THREAD

**Discussion by anschl** [eclipse-tractusx/tractusx-edc#1224] [General]
We currently read data from the EDC like this:
1. The REST client checks if there is an EDR for the asset
2. The client creates the EDR if none was found
3. The client calls `/aas/request` and returns the payload

This has a chance of creating the EDR twice if two requests come in with an unlucky timing, leading to an error in step 3. Do I need to take care of that or is there a solution for this in the EDC?

**Comment by anschl** [comment]
Summary of the answer I got on the call:
Conflicts are possible, however it should only be a problem while testing, as creating multiple EDRs rapidly is not expected in production use. EDRs can be compared to opening a socket connection in this regard.

**Reply by florianrusch-zf** [reply]
Hi, could you or someone quickly explain why it's not expected, that in production rapidly multiple EDRs will be created?

**Reply by anschl** [reply]
I will leave the answer to someone more knowledgeable than I am. In case you need the answer in a reasonable time, I recommend joining the [weekly call](https://github.com/eclipse-tractusx/tractusx-edc/discussions/1057) and ask there.
